packages/core/src/metric-summary.ts

```diff
@@ -0,0 +1,56 @@
+import type { MeasurementUnit, Primitive, SpanMetricSummaryAggregator } from '@sentry/types';
+import type { MetricSpanSummary } from '@sentry/types';
+import { dropUndefinedKeys } from '@sentry/utils';
+
+/** */
+export class MetricSummaryAggregator implements SpanMetricSummaryAggregator {
+  private readonly _measurements: Map<string, MetricSpanSummary>;
+
+  public constructor() {
+    this._measurements = new Map<string, MetricSpanSummary>();
+  }
+
+  /** @inheritdoc */
+  public add(
+    metricType: 'c' | 'g' | 's' | 'd',
+    name: string,
+    value: number,
+    unit?: MeasurementUnit | undefined,
+    tags?: Record<string, Primitive> | undefined,
+  ): void {
+    const exportKey = `${metricType}:${name}@${unit}`;
+    const bucketKey = `${exportKey}\n${JSON.stringify(tags)}`;
+
+    const summary = this._measurements.get(bucketKey);
+
+    if (summary) {
+      this._measurements.set(bucketKey, {
+        min: Math.min(summary.min, value),
+        max: Math.max(summary.max, value),
+        count: (summary.count += 1),
+        sum: (summary.sum += value),
+        tags: summary.tags,
+      });
+    } else {
```

Comment:
If the metric type is `set`, you set the value to `0.0`.

Reply:
See https://github.com/getsentry/sentry-python/blob/c8e917205d77e143c68c0cda4df877167d402830/sentry_sdk/metrics.py#L545-L554C33, there is a special handling for sets, even if the value is not a string.